packages/block-library/src/social-links/social-link/icon-picker.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+
+const IconPicker = ( { icons, onClick } ) => {
+	return (
+		<div className="wp-block-social-link__icon-picker">
+			{
+				icons.map( ( icon ) => {
+					const classes = classNames( 'wp-social-icon', `wp-social-icon-${ icon }` );
+					return (
+						<Button
+							key={ icon }
+							className={ classes }
+							data-icon={ icon }
+							onClick={ ( evt ) => onClick( evt.target.dataset.icon ) }
```

Comment:
Is `data-icon` really necessary here, you can pass the name of the icon directly to the `onClick` handler?

Reply:
I'll try to refactor, I was struggling passing it in and need to circle back on this one.

Reply:
Fixed in 608dfaff0
I'm really not sure what I was doing the first time.